libs/coroutines/include/hpx/coroutines/detail/coroutine_self.hpp

```diff
@@ -129,67 +117,33 @@ namespace hpx { namespace threads { namespace coroutines { namespace detail {
             return tmp;
         }
 
-        thread_id_type get_thread_id() const
-        {
-            HPX_ASSERT(m_pimpl);
-            return m_pimpl->get_thread_id();
-        }
+        virtual ~coroutine_self() = default;
```

Comment:
Do you know if making all these functions virtual adds any noticeable overhead?

Reply:
I have not seen any significant slowdown from those, but I have not done any exhaustive measurements,